### PR TITLE
fix: Rule to match entire body (When body is not an object)

### PR DESCRIPTION
**Pull request by bykof**
**Parent issue** 

Closes #https://github.com/mockoon/mockoon/issues/271

**Technical implementation details**

Nothing really complicated

### CODE REVIEW

**Comment by bykof** [comment]
I will also update mockoon to use the new rule "body_plain"

**Comment by 255kb** [comment]
Thank you @bykof for this contribution. 
Could you please comment on the main issue so I can assign it to you? 

Concerning the implementation, instead of a new `target` option, I would have rather seen an empty `modifier` to check against the entire body. This would be more in line with the `body` templating helper which accepts a `path` parameter `{{body 'object.path}}` or nothing to get the entire body `{{body}}`. What do you think?

**Comment by bykof** [comment]
Hey @255kb 

> Could you please comment on the main issue so I can assign it to you?

done.

> Concerning the implementation, instead of a new `target` option, I would have rather seen an empty `modifier` to check against the entire body. This would be more in line with the `body` templating helper which accepts a `path` parameter `{{body 'object.path}}` or nothing to get the entire body `{{body}}`. What do you think?

Yeah that sounds really good, I will change it.

**Comment by 255kb** [comment]
> > Concerning the implementation, instead of a new `target` option, I would have rather seen an empty `modifier` to check against the entire body. This would be more in line with the `body` templating helper which accepts a `path` parameter `{{body 'object.path}}` or nothing to get the entire body `{{body}}`. What do you think?
> 
> Yeah that sounds really good, I will change it.
👍

**Comment by bykof** [comment]
I think this should work right now. Please check my friend

File: src/libs/response-rules-interpreter.ts [hunk 1]
```diff
@@ -13,7 +13,9 @@ import {
  * Get the first route response for which at least one rule is fulfilled.
  */
 export class ResponseRulesInterpreter {
-  private targets: { [key in Exclude<ResponseRuleTargets, 'header'>]: any };
+  private targets: {
+    [key in Exclude<ResponseRuleTargets, 'header' | 'body_plain'>]: any;
```
Comment (255kb) [inline src/libs/response-rules-interpreter.ts:17]: Do we still need to exclude `body_plain`?
```suggestion
    [key in Exclude<ResponseRuleTargets, 'header'>]: any;
```

**Review by 255kb — CHANGES_REQUESTED** [review]

File: src/libs/response-rules-interpreter.ts (hunk 1 above)
Comment (bykof) [reply src/libs/response-rules-interpreter.ts]: Yup sorry, I forgot

**Review by 255kb — APPROVED** [review]

**Comment by 255kb** [comment]
@bykof lib is published on NPM v1.1.1
Thanks :)

**Comment by bykof** [comment]
Thank you my friend! I am currently testing it locally :)

**Comment by 255kb** [comment]
Do not hesitate if you need help with the tests

**Pull request merged**